WordPress/gutenberg
# Mobile bottom sheet component

## Description


This PR introduces a `BottomSheet` component to be used on mobile. It's a special use-case of the `react-native-modal` component implementing our design and animations.

It has props for the header's left and right buttons and title, and the `onClose` action.

It's firstly implemented as the Image Settings interface for the Image block.

![bottom-sheet](https://user-images.githubusercontent.com/9772967/52041951-21fcec80-253c-11e9-96c0-398844de84d9.gif)

To test:
- Checkout the `gutenberg-mobile` side PR (https://github.com/wordpress-mobile/gutenberg-mobile/pull/529).
- Build and run the example project.
- Select an image block with image.
- Press the settings button (dented wheel) in the inline toolbar.
- Check that the bottom sheet appears.
- Check that the bottom sheet dismiss with when 
  - Tapping on the `Done` button.
  - Tapping on the background.
  - Sliding it down.
